web/static.go

```diff
@@ -73,7 +76,25 @@ func root(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Cache-Control", "no-cache, max-age=31556926, public")
 
 	staticDir, _ := fileutils.FindDir(model.ClientDir)
-	http.ServeFile(w, r, filepath.Join(staticDir, "root.html"))
+	contents, err := os.ReadFile(filepath.Join(staticDir, "root.html"))
+	if err != nil {
+		// TODO: may be we should return an error page here.
+		return
+	}
+
+	newHTML := string(contents)
```

Comment:
There is no need to convert this to a string, all methods you call also exist for `[]byte`.